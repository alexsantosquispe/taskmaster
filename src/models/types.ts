import type { Control, FieldValues } from 'react-hook-form';

import type { ReactNode } from 'react';

export interface SvgIconProps {
  className?: string;
}

export type TabItem = {
  id: string;
  label: string;
  icon?: ReactNode;
};

export const PRIORITY_TYPES = {
  LOW: 'Low',
  MEDIUM: 'Medium',
  HIGH: 'High'
} as const;

export type PriorityType = (typeof PRIORITY_TYPES)[keyof typeof PRIORITY_TYPES];

export const STATUS_TYPES = {
  BACKLOG: 'backlog',
  ON_HOLD: 'on-hold',
  IN_PROGRESS: 'in-progress',
  REVIEW: 'review',
  DONE: 'done'
} as const;

export type StatusType = (typeof STATUS_TYPES)[keyof typeof STATUS_TYPES];

export interface UserType {
  id: string;
  name: string;
  avatarUrl: string;
}

export interface NavBarLinkItemBase {
  id: string;
  label: string;
  path: string;
}

export interface NavBarLinkSubItem extends NavBarLinkItemBase {
  color: string;
}

export interface NavBarLinkItem extends NavBarLinkItemBase {
  icon?: ReactNode;
  subItems?: NavBarLinkSubItem[];
}

export type TaskType = {
  id: string;
  title: string;
  description: string;
  status: StatusType;
  priority: PriorityType;
  progress: number;
  createdAt: string; //datetime: 'YYYY-MM-DDThh:mm:ss-TZ'
  lastUpdate: string; //datetime
  assignee?: UserType;
};

export interface InputProps {
  label: string;
  name: string;
  control: Control<FieldValues>;
  placeholder?: string;
  type?: string;
  errorMessage?: string;
  isDisabled?: boolean;
  isRequired?: boolean;
}

export type TextAreaProps = Omit<InputProps, 'type'> & {
  rows?: number;
  cols?: number;
};
