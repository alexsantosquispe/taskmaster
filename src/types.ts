export interface SvgIconProps {
  className?: string;
}

export type LSValuesTypes = string | number | boolean;

export type TabItem = {
  id: string;
  label: string;
};
