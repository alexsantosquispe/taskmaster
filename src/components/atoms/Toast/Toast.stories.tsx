import type { Meta, StoryObj } from '@storybook/react';

import { TOAST_TYPES } from '@/models/types';
import { Toast } from './Toast';

const meta: Meta<typeof Toast> = {
  title: 'Atoms/Toast',
  component: Toast,
  parameters: {
    layout: 'centered'
  }
};

export default meta;

type Story = StoryObj<typeof Toast>;

export const Default: Story = () => {
  return (
    <div className="flex flex-col gap-8">
      <Toast
        id="1"
        title="Default toast"
        message="message"
        type={TOAST_TYPES.DEFAULT}
        onRemove={() => {}}
      />
      <Toast
        id="2"
        title="Success toast"
        message="message"
        type={TOAST_TYPES.SUCCESS}
        onRemove={() => {}}
      />
      <Toast
        id="3"
        title="Error toast"
        message="message"
        type={TOAST_TYPES.ERROR}
        onRemove={() => {}}
      />
    </div>
  );
};

Default.args = {};
