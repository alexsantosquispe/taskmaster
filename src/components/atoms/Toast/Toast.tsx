import { CloseIcon } from '@/icons';
import { TOAST_TYPES, type ToastMessage } from '@/models/types';
import cn from 'clsx';
import { useEffect } from 'react';
import { twMerge } from 'tailwind-merge';
import { IconButton } from '../IconButton/IconButton';

interface ToastProps extends ToastMessage {
  onRemove: (toastId: string) => void;
  className?: string;
}

export const Toast = ({
  id,
  title,
  message,
  onRemove,
  type = TOAST_TYPES.DEFAULT,
  duration,
  className
}: ToastProps) => {
  useEffect(() => {
    if (!duration) return;
    const timer = setTimeout(() => onRemove(id), duration);
    return () => clearTimeout(timer);
  }, [duration, id]);

  return (
    <div
      data-testid={`toast-${id}`}
      className={twMerge(
        'flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg',
        cn({
          'text-primary dark:bg-primary border-neutral-500 bg-white dark:border-white/30 dark:text-white':
            type === TOAST_TYPES.DEFAULT,
          'border-green-600 bg-green-50 text-green-700 dark:border-green-600 dark:bg-green-950 dark:text-green-500':
            type === TOAST_TYPES.SUCCESS,
          'border-rose-600 bg-rose-50 text-rose-700 dark:border-rose-600 dark:bg-rose-950 dark:text-rose-500':
            type === TOAST_TYPES.ERROR
        }),
        className
      )}
    >
      <div className="flex items-start justify-between">
        <span className="font-bold">{title}</span>
        <IconButton
          icon={<CloseIcon className="size-4" />}
          onClick={() => onRemove(id)}
          ariaLabel="Close button"
          className={twMerge(
            '-mt-1.5 -mr-1.5 rounded-full p-1',
            cn({
              'hover:bg-green-500/15 hover:text-green-700 dark:hover:text-green-500':
                type === TOAST_TYPES.SUCCESS,
              'hover:bg-rose-100 hover:text-rose-700 dark:hover:text-rose-500':
                type === TOAST_TYPES.ERROR
            })
          )}
          isDefault={false}
        />
      </div>

      <p className="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300">
        {message}
      </p>
    </div>
  );
};
