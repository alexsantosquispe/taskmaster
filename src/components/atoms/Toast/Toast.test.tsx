import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import { TOAST_TYPES } from '@/models/types';
import { Toast } from './Toast';

describe('Toast', () => {
  const onRemoveMock = jest.fn();
  const props = {
    id: 'test',
    title: 'test',
    message: 'test',
    onRemove: onRemoveMock
  };

  describe('styles', () => {
    it('should render the component correctly default, error, and success types', () => {
      let component = render(<Toast {...props} />);
      expect(component).toMatchSnapshot();

      component = render(<Toast {...props} type={TOAST_TYPES.SUCCESS} />);
      expect(component).toMatchSnapshot();

      component = render(<Toast {...props} type={TOAST_TYPES.ERROR} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('should remove toast on close button click', () => {
      render(<Toast {...props} />);

      const closeButton = screen.getByRole('button');

      expect(closeButton).toBeInTheDocument();

      fireEvent.click(closeButton);

      expect(onRemoveMock).toHaveBeenCalledTimes(1);
    });

    it('should remove toast after duration', () => {
      render(<Toast {...props} duration={1000} />);

      expect(screen.getByRole('button')).toBeInTheDocument();

      expect(onRemoveMock).toHaveBeenCalledTimes(1);
    });
  });
});
