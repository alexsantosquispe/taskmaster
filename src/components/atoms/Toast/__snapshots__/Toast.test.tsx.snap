// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Toast styles should render the component correctly default, error, and success types 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg text-primary dark:bg-primary border-neutral-500 bg-white dark:border-white/30 dark:text-white"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:text-primary hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 dark:hover:text-white -mt-1.5 -mr-1.5 rounded-full p-1"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg text-primary dark:bg-primary border-neutral-500 bg-white dark:border-white/30 dark:text-white"
      data-testid="toast-test"
    >
      <div
        class="flex items-start justify-between"
      >
        <span
          class="font-bold"
        >
          test
        </span>
        <div
          class="relative flex items-center justify-center"
          data-testid="tooltip-wrapper"
        >
          <button
            aria-label="Close button"
            class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:text-primary hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 dark:hover:text-white -mt-1.5 -mr-1.5 rounded-full p-1"
            type="button"
          >
            <svg
              class="size-4"
              data-testid="close-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
      >
        test
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toast styles should render the component correctly default, error, and success types 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg text-primary dark:bg-primary border-neutral-500 bg-white dark:border-white/30 dark:text-white"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:text-primary hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 dark:hover:text-white -mt-1.5 -mr-1.5 rounded-full p-1"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg border-green-600 bg-green-50 text-green-700 dark:border-green-600 dark:bg-green-950 dark:text-green-500"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:cursor-pointer dark:hover:bg-white/15 -mt-1.5 -mr-1.5 rounded-full p-1 hover:bg-green-500/15 hover:text-green-700 dark:hover:text-green-500"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg border-green-600 bg-green-50 text-green-700 dark:border-green-600 dark:bg-green-950 dark:text-green-500"
      data-testid="toast-test"
    >
      <div
        class="flex items-start justify-between"
      >
        <span
          class="font-bold"
        >
          test
        </span>
        <div
          class="relative flex items-center justify-center"
          data-testid="tooltip-wrapper"
        >
          <button
            aria-label="Close button"
            class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:cursor-pointer dark:hover:bg-white/15 -mt-1.5 -mr-1.5 rounded-full p-1 hover:bg-green-500/15 hover:text-green-700 dark:hover:text-green-500"
            type="button"
          >
            <svg
              class="size-4"
              data-testid="close-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
      >
        test
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toast styles should render the component correctly default, error, and success types 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg text-primary dark:bg-primary border-neutral-500 bg-white dark:border-white/30 dark:text-white"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:text-primary hover:cursor-pointer hover:bg-neutral-100 dark:hover:bg-white/15 dark:hover:text-white -mt-1.5 -mr-1.5 rounded-full p-1"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg border-green-600 bg-green-50 text-green-700 dark:border-green-600 dark:bg-green-950 dark:text-green-500"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:cursor-pointer dark:hover:bg-white/15 -mt-1.5 -mr-1.5 rounded-full p-1 hover:bg-green-500/15 hover:text-green-700 dark:hover:text-green-500"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
    <div>
      <div
        class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg border-rose-600 bg-rose-50 text-rose-700 dark:border-rose-600 dark:bg-rose-950 dark:text-rose-500"
        data-testid="toast-test"
      >
        <div
          class="flex items-start justify-between"
        >
          <span
            class="font-bold"
          >
            test
          </span>
          <div
            class="relative flex items-center justify-center"
            data-testid="tooltip-wrapper"
          >
            <button
              aria-label="Close button"
              class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:cursor-pointer dark:hover:bg-white/15 -mt-1.5 -mr-1.5 rounded-full p-1 hover:bg-rose-100 hover:text-rose-700 dark:hover:text-rose-500"
              type="button"
            >
              <svg
                class="size-4"
                data-testid="close-icon"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </button>
          </div>
        </div>
        <p
          class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
        >
          test
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex max-h-[4.375rem] w-[22rem] flex-col gap-1 rounded-lg border p-3 text-sm shadow-lg backdrop-blur-lg border-rose-600 bg-rose-50 text-rose-700 dark:border-rose-600 dark:bg-rose-950 dark:text-rose-500"
      data-testid="toast-test"
    >
      <div
        class="flex items-start justify-between"
      >
        <span
          class="font-bold"
        >
          test
        </span>
        <div
          class="relative flex items-center justify-center"
          data-testid="tooltip-wrapper"
        >
          <button
            aria-label="Close button"
            class="text-primary/80 flex w-fit items-center justify-center transition-colors duration-150 ease-in-out dark:text-white/70 hover:cursor-pointer dark:hover:bg-white/15 -mt-1.5 -mr-1.5 rounded-full p-1 hover:bg-rose-100 hover:text-rose-700 dark:hover:text-rose-500"
            type="button"
          >
            <svg
              class="size-4"
              data-testid="close-icon"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
          </button>
        </div>
      </div>
      <p
        class="font-regular line-clamp-1 text-neutral-700 dark:text-neutral-300"
      >
        test
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
